internal/provider/resource_aci_pim_route_map_entry.go

```diff
@@ -0,0 +1,595 @@
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"parent_dn": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "The distinquised name (DN) of the parent object.",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"action": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				Validators: []validator.String{
+					stringvalidator.OneOf("deny", "permit"),
+				},
+				MarkdownDescription: `route action.`,
+			},
+			"annotation": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				Default:             stringdefault.StaticString(globalAnnotation),
+				MarkdownDescription: `The annotation of the Pim Route Map Entry object.`,
+			},
+			"description": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				MarkdownDescription: `The description of the Pim Route Map Entry object.`,
+			},
+			"grp": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				MarkdownDescription: `Multicast group ip/prefix.`,
+			},
+			"name": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				MarkdownDescription: `The name of the Pim Route Map Entry object.`,
+			},
+			"name_alias": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				MarkdownDescription: `The name alias of the Pim Route Map Entry object.`,
+			},
+			"order": schema.StringAttribute{
+				Required: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+					stringplanmodifier.RequiresReplace(),
+				},
+				MarkdownDescription: `PIM route map entry order.`,
+			},
+			"rp": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				MarkdownDescription: `Multicast RP Ip.`,
+			},
+			"src": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				MarkdownDescription: `Multicast Source Ip.`,
+			},
+			"annotations": schema.SetNestedAttribute{
+				MarkdownDescription: ``,
+				Optional:            true,
+				Computed:            true,
+				PlanModifiers: []planmodifier.Set{
```

Comment:
Under what scenario is []planmodifier.Set triggered? Is this required if all the nested attributes of annotations have their own plan modifier?

Reply:
No this is required for when the config is not provided but we do want to make that config ( if configured on apic ) to be displayed in state. It would behave similar to the previous computed behaviour.